adapters/risemediatech/risemediatech.go

```diff
@@ -0,0 +1,162 @@
+package risemediatech
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v3/adapters"
+	"github.com/prebid/prebid-server/v3/config"
+	"github.com/prebid/prebid-server/v3/errortypes"
+	"github.com/prebid/prebid-server/v3/openrtb_ext"
+	"github.com/prebid/prebid-server/v3/util/iterutil"
+	"github.com/prebid/prebid-server/v3/util/jsonutil"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+func Builder(bidderName openrtb_ext.BidderName, cfg config.Adapter, server config.Server) (adapters.Bidder, error) {
+	return &adapter{endpoint: cfg.Endpoint}, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errs []error
+	var validImps []openrtb2.Imp
+	var setTestMode bool
+
+	for imp := range iterutil.SlicePointerValues(request.Imp) {
+		impExt, err := parseImpExt(imp.Ext)
+		if err != nil {
+			errs = append(errs, &errortypes.BadInput{Message: fmt.Sprintf("impID %s: %v", imp.ID, err)})
+			continue
+		}
+
+		if imp.Banner != nil {
+			if imp.Banner.W == nil || imp.Banner.H == nil || *imp.Banner.W == 0 || *imp.Banner.H == 0 {
+				errs = append(errs, &errortypes.BadInput{Message: fmt.Sprintf("impID %s: invalid banner dimensions", imp.ID)})
+				continue
+			}
+		}
+
+		if imp.Video != nil {
+			if len(imp.Video.MIMEs) == 0 {
+				errs = append(errs, &errortypes.BadInput{Message: fmt.Sprintf("impID %s: missing or empty video.mimes", imp.ID)})
+				continue
+			}
+			if imp.Video.W == nil || imp.Video.H == nil || *imp.Video.W == 0 || *imp.Video.H == 0 {
+				errs = append(errs, &errortypes.BadInput{Message: fmt.Sprintf("impID %s: missing or invalid video width/height", imp.ID)})
+				continue
+			}
+		}
+
+		if impExt.BidFloor > 0 {
+			imp.BidFloor = impExt.BidFloor
+		}
+
+		if impExt.TestMode == 1 {
+			setTestMode = true
+		}
+
+		validImps = append(validImps, *imp)
+	}
+
+	if len(validImps) == 0 {
+		return nil, append(errs, &errortypes.BadInput{Message: "no valid impressions"})
+	}
+
+	modifiedRequest := *request
+	modifiedRequest.Imp = validImps
+	if setTestMode {
+		modifiedRequest.Test = 1
+	}
+
+	reqJSON, err := jsonutil.Marshal(modifiedRequest)
+	if err != nil {
+		return nil, append(errs, err)
+	}
+
+	headers := http.Header{}
+	headers.Set("Content-Type", "application/json;charset=utf-8")
+	headers.Set("Accept", "application/json")
+	headers.Set("User-Agent", "prebid-server")
+	headers.Set("X-Prebid", "true")
+
+	return []*adapters.RequestData{
+		{
+			Method:  "POST",
+			Uri:     a.endpoint,
+			Body:    reqJSON,
+			Headers: headers,
+			ImpIDs:  extractImpIDs(validImps),
```

Comment:
Please delete the helper function `extractImpIDs` and instead use the core helper function `openrtb_ext.GetImpIDs`.